exporter/prometheusremotewriteexporter/wal.go

```diff
@@ -41,7 +44,35 @@ type prweWAL struct {
 	wWALIndex uint64
 }
 
-func newWAL(walConfig *walConfig, exportSink func(context.Context, []*prompb.WriteRequest) []error) (*prweWAL, error) {
+const (
+	defaultWALBufferSize        = 300
+	defaultWALTruncateFrequency = 1 * time.Minute
+)
+
+type WALConfig struct {
+	// Note: These variable names are meant to closely mirror what Prometheus' WAL uses for field names per
+	// https://docs.google.com/document/d/1cCcoFgjDFwU2n823tKuMvrIhzHty4UDyn0IcfUHiyyI/edit#heading=h.mlf37ibqjgov
+	// but also we are using underscores "_" instead of dashes "-".
+	Directory         string        `mapstructure:"directory"`
+	BufferSize        int           `mapstructure:"buffer_size"`
```

Comment:
Good question.  I didn't write this code so I'm not sure whether it's a case of that document having changed since this was written or the code having evolved and the comment not being updated.  The doc does have some discussion regarding the way the timestamp used is obtained being ugly, and it depends on querying the stored metrics that we don't have here.  I'm going to delete the comment as it is misleading.